packages/path_provider/path_provider_android/test/path_provider_android_test.dart

```diff
@@ -98,13 +98,9 @@ void main() {
       });
     } // end of for-loop
 
-    test('getDownloadsPath fails', () async {
-      try {
-        await pathProvider.getDownloadsPath();
-        fail('should throw UnsupportedError');
-      } catch (e) {
-        expect(e, isUnsupportedError);
-      }
+    test('getDownloadsPath succeeds', () async {
```

Comment:
I missed this before; you're only testing one of three different codepaths. Please test both the null and empty platform channel conditions that the code is intended to handle.

See discussion in https://github.com/flutter/packages/pull/4317, which is another PR intended to add this functionality, but which hasn't been completed.

Reply:
Added the tests, please have a look. @stuartmorgan